### DISCUSSION TITLE
Problem with R and Gama image installation and Error when trying to run simulation on ChargEval

### DISCUSSION THREAD

**Discussion by faisalanshory** [chintanp/ChargEval#19] [Q&A]
Hello, I am trying to redeploy the ChargEval application on https://chargeval.my.id/evides/ by following the available documentation. I have several questions regarding the process of deployment of application:

1. How do I create R and Gama images in the AWS EC2 server?
2. I have successfully deployed Evides and Resview components on Amazon EC2 Server, but after I put new EVSEs and then try to run the simulation, the screen gets blurry (Image 1) and then when I checked the server I found an error as shown in Image 2. I am not sure what is the error trying to tell. Do you have any suggestions to solve this problem?

My assumption is I think my failure to create R and Gama images causes the problem in number 2, but somehow I could not find proper references to solve it. Please help me if you know the solution related to my questions.

Thanks in advance.

Image 1:
![Image 1](https://user-images.githubusercontent.com/57836379/158845684-e95fbfe7-eef6-4ba5-89d4-1a65d14a8841.jpeg)
Image 2:
![Image 2](https://user-images.githubusercontent.com/57836379/158845692-2e900aac-2373-4337-a6d8-cd0498284c6e.jpeg)

**Comment by chintanp** [comment]
@faisalanshory Please use this repo in future for issues: https://github.com/s-t-lab/ChargEval. This is the lab repo and will be the version getting fixes and updates from future students in the lab. 

About the current issues: 

1. The R and GAMA AMIs can be created from the files in the repo : tripgen (https://github.com/chintanp/wsdot_evse_update_states) and eviabm (https://github.com/chintanp/evi-abm). I realize now that this hasn't been adequately documented. The easier way for me would be to make a copy of the current AMIs public and you can modify those for your setup. The only item that would need changing is the DB details. I will do this later today. You can also do this yourself from the relevant codes. SIMMAN then needs the AMI IDs for these AMIs to launch an instance of. 
2. I am unsure about the cause of the second error, just from this. But this has nothing to do with the R/GAMA images being absent. The workflow is -> an analysis request is inserted into the database. The SIMMAN then picks up inserts and launches new instances. So the absence of AMIs would mean that the simulation would not happen; however, the analysis request should still be inserted into the database. So, the error is happening when it is trying to insert into the database. You can debug the "query_user" at which step I believe the error is happening, and see what the values being inserted are, vs what the database expects. The datatype should match and there should be a value in the column if it is not null. Before running this on EC2, you should run this locally. You can use RStudio in EC2 too for debugging and place breakpoints in the relevant code sections and execute step by step. 

Hope this helps.

**Reply by faisalanshory** [reply]
Thank you, I will try to follow your instruction and will update the output later

**Reply by faisalanshory** [reply]
Hello Mr @chintanp, I have already opened a discussion in the new repository that you mean (https://github.com/s-t-lab/ChargEval/discussions/3). Please check it out if you are available.
